backend/app/services/dividend_computation_generation.rb

```diff
@@ -92,11 +124,39 @@ def generate_common_dividends
           total_amount_in_usd: dividend_usd,
           investment_amount_cents: convertible.amount_in_cents,
         }
-        computation.dividend_computation_outputs.create!(attrs)
+        @common_dividend_outputs << attrs
         @common_dividend_total += dividend_usd
       end
     end
 
+    def save_preferred_dividends
+      @preferred_dividend_outputs.each do |attrs|
+        computation.dividend_computation_outputs.create!(attrs)
+      end
+    end
+
+    def save_common_dividends
+      @common_dividend_outputs.each do |attrs|
+        if attrs[:type] == :share_holding_update
+          find_attrs = {
+            company_investor_id: attrs[:company_investor_id],
+            share_class: attrs[:share_class],
+            number_of_shares: attrs[:number_of_shares],
+          }
+          output = computation.dividend_computation_outputs.find_by(find_attrs)
+          if output
+            output.dividend_amount_in_usd = attrs[:dividend_amount_in_usd]
+            output.qualified_dividend_amount_usd += attrs[:qualified_dividend_amount_usd]
+            output.total_amount_in_usd += attrs[:total_amount_in_usd]
+            output.save!
+          end
```

Comment:
```suggestion
          output = computation.dividend_computation_outputs.find_by!(find_attrs)
          output.dividend_amount_in_usd = attrs[:dividend_amount_in_usd]
          output.qualified_dividend_amount_usd += attrs[:qualified_dividend_amount_usd]
          output.total_amount_in_usd += attrs[:total_amount_in_usd]
          output.save!
```

A change I know but it makes sense.